src/commands/play.rs

```diff
@@ -190,19 +190,28 @@ async fn create_queued_embed(
     embed
 }
 
-async fn enqueue_song(call: &Arc<Mutex<Call>>, query: String, is_url: bool, flag: &PlayFlag) {
-    let source = if is_url {
-        YouTubeRestartable::ytdl(query, true).await.unwrap()
+async fn enqueue_song(call: &Arc<Mutex<Call>>, query: String, is_url: bool, flag: &PlayMode) {
+    let source_return = if is_url {
+        YouTubeRestartable::ytdl(query, true).await
     } else {
-        YouTubeRestartable::ytdl_search(query, true).await.unwrap()
+        YouTubeRestartable::ytdl_search(query, true).await
+    };
+
+    // safeguard against ytdl dying on a private / deleted video and killing the playlist
+    let source = match source_return {
+        Ok(source) => source,
+        Err(error) => {
+            println!("{}", error);
+            return;
+        }
     };
```

Comment:
Good catch! 🚀

Reply:
Indeed! But maybe we could add some text to explain what happened?

Reply:
@afonsojramos You mean a Discord message to inform the user?
Something like **"An error occurred while trying to queue track `XXXXX`! This video might be privated."**?

Reply:
The println right now does not make much sense, as this prints to the server side and we don't have server logs (which in the future could be something we look into, using a logger).

Sending a message for each private track to the discord server might be too much. I would say just fail silently (return only).

Reply:
I actually would leave the print so we can easily replace with the logger in the future.